platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/XMLUtils.java

```diff
@@ -195,4 +232,41 @@ private static void transformation(Source sourceXml, TransformerProperties trans
                     .setContextClassLoader(tccl);
         }
     }
+
+    // This class is used with the processElements method. It works in conjunction with the
+    // #processElementFunction. Inside the function, set the value of the ProcessingResult and
+    // call done() to stop processing the XML document.
+    public static class ProcessingResult<T> {
+        boolean isDone = false;
+
+        T value;
+
+        public ProcessingResult(T value) {
+            this.value = value;
+        }
+
+        public ProcessingResult() {
+        }
+
+        public T getValue() {
+            return value;
+        }
+
+        public void setValue(T value) {
+            this.value = value;
+        }
+
+        public boolean isDone() {
+            return isDone;
+        }
+
+        public void setDone(boolean done) {
```

Comment:
I don't see the advantage of having both a `setDone()` method and a convenience `done()` method.